app/shared/taginput/taginput.component.ts

```diff
@@ -0,0 +1,156 @@
+﻿import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { User } from '../../core/user/user.model';
+import { UserService } from '../../core/user/user.service';
+
+import {
+  Validators,
+  FormGroup,
+  FormControl
+} from '@angular/forms';
+
+@Component({
+  selector: 'hip-taginput',
+  templateUrl: './app/shared/taginput/taginput.component.html'
+})
+export class TagInputComponent {
+
+  public errorMessage: any;       // Handling error message
+  public names: string[] = [];    // AutoComplete List
+  @Input() role: string;          // User role Passed dynamically
+  @Input() users: number[];         // List of Users updated
```

Comment:
I think it is better to input a User Array instead of just ids in the Component. Doing so, we don't need to extract the UserIds when we fetch the Users from the server and have to lists of Users we have to keep consistent.